plugins/org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/builderState/MarkerUpdaterImpl.java

```diff
@@ -65,12 +65,17 @@ private void processDelta(Delta delta, /* @Nullable */ ResourceSet resourceSet,
 		IMarkerContributor markerContributor = getMarkerContributor(uri);
 		CheckMode normalAndFastMode = CheckMode.NORMAL_AND_FAST;
 
-		for (Pair<IStorage, IProject> pair : mapper.getStorages(uri)) {
+		for (Pair<IStorage, IProject> pair : mapper.getStorages(uri, false)) {
 			if (monitor.isCanceled()) {
 				throw new OperationCanceledException();
 			}
 			if (pair.getFirst() instanceof IFile) {
 				IFile file = (IFile) pair.getFirst();
+				
```

Comment:
We should try to identify these to be skipped resources without touching the mapper.
What is file.getProject() returning in your particular case?
Also what is file.isReadOnly() returning?